static/app/views/dashboardsV2/widget/metricWidget/metricSelectField.tsx

```diff
@@ -0,0 +1,106 @@
+import React from 'react';
+import {components, OptionProps} from 'react-select';
+import styled from '@emotion/styled';
+
+import SelectControl from 'app/components/forms/selectControl';
+import Highlight from 'app/components/highlight';
+import Tooltip from 'app/components/tooltip';
+import {t} from 'app/locale';
+import SelectField from 'app/views/settings/components/forms/selectField';
+
+import {MetricMeta} from './types';
+
+type Props = {
+  metricMetas: MetricMeta[];
+  onChange: <F extends keyof Pick<Props, 'metricMeta' | 'aggregation'>>(
+    field: F,
+    value: Props[F]
+  ) => void;
+  aggregation?: MetricMeta['operations'][0];
+  metricMeta?: MetricMeta;
+};
+
+function MetricSelectField({metricMetas, metricMeta, aggregation, onChange}: Props) {
+  const operations = metricMeta?.operations ?? [];
+  return (
+    <Wrapper>
+      <StyledSelectField
+        name="metric"
+        choices={metricMetas.map(m => [m.name, m.name])}
+        placeholder={t('Select metric')}
+        onChange={v => {
+          const newMetric = metricMetas.find(m => m.name === v);
```

Comment:
Should we give these single-letter variables a proper name?

Reply:
The reason I decided to use single-letters was because of conflicts but I can rename them 😉